pkg/controller/volume/protectionutil/utils_test.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package protectionutil
+
+import (
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+const (
+	defaultNS      = "default"
+	defaultPVCName = "pvc1"
+	defaultPVName  = "pv1"
+)
+
+type TestCase struct {
+	name      string
+	obj       metav1.Object
+	finalizer string
+	result    bool
+}
+
+func pvc() *v1.PersistentVolumeClaim {
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultPVCName,
+			Namespace: defaultNS,
+		},
+	}
+}
+
+func pv() *v1.PersistentVolume {
+	return &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: defaultPVName,
+		},
+	}
+}
+
+func TestIsDeletionCandidateNoDeleteTime(t *testing.T) {
+
+	tests := []TestCase{
+		{
+			name:      "pv does not have delete time",
```

Comment:
how about PV and PVC has deletion timestamp? I think we have four combination of having deletion timestamp and  finalizers or not.

Reply:
> how about PV and PVC has deletion timestamp? I think we have four combination of having deletion timestamp and finalizers or not.

Thank you for your advice. I have add some other uts